### DISCUSSION TITLE
use_space_char parameter not working

### DISCUSSION THREAD

**Discussion by ThanhLV97** [PaddlePaddle/PaddleOCR#6651] [General]
use_space_char parameter not working when training a custom text recognition. Can you please tell me why? and How to resolve this problem? Tks

**Comment by tink2123** [comment]
"use_space_char" automatically adds a space character to the dictionary:
https://github.com/PaddlePaddle/PaddleOCR/blob/0438b4066469be7a37bba64ad7af7eeb041063f7/ppocr/data/imaug/label_ops.py#L123-L124

> not working when training a custom text recognition.

Can you elaborate on what went wrong while training?  Why do you say it doesn't work?

**Reply by ThanhLV97** [reply]
Thank for your reply. I has checked all parameters which are corrected according to your readme file and the most importance I set use_space_char = True while training. However in the dataset I do not have any annotation for space charracter. I  meaning space charracter will be affected in inference process. Do you think that's right?

**Reply by LDOUBLEV** [reply]
If you expect to recognize spaces and spaces are annotated, you need to set use_space_char to True during training.
Otherwise, it is not needed.
'https://github.com/PaddlePaddle/PaddleOCR/blob/0438b4066469be7a37bba64ad7af7eeb041063f7/configs/rec/ch_PP-OCRv2/ch_PP-OCRv2_rec.yml#L19'

**Reply by trucdtt** [reply]
@LDOUBLEV I have the same problems. I tried to train English text recognition model followed by readme file ([https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.5/doc/doc_en/recognition_en.md](url)). With the ICDAR2015 dataset and  *en_PP-OCRv3_rec.yml* config, predicted results are still no space.

**Reply by LDOUBLEV** [reply]
> @LDOUBLEV I have the same problems. I tried to train English text recognition model followed by readme file ([https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.5/doc/doc_en/recognition_en.md](url)). With the ICDAR2015 dataset and _en_PP-OCRv3_rec.yml_ config, predicted results are still no space.

Because the space character is not one of character list of the icdar2015 dataset.

**Reply by trucdtt** [reply]
There has been confusion about annotating space in ICDAR 2015 and ICDAR 2019. First, I trained with ICDAR 2019 and *en_PP-OCRv3_rec.yml* config, so the predicted results are no space. After that, I tried to re-train with ICDAR 2015, text after recognition has space between text. I checked ground truth in 2 datasets and found that space characters are only annotated in ICDAR 2015.

**Reply by ThanhLV97** [reply]
> I have a sample confusion. Can paddle's team confirm that ? @LDOUBLEV  thanks
